clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2870,6 +2983,24 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
     return true;
   }
 
+  void handleSyclKernelHandlerType(QualType Ty) {
+    // The compiler generated kernel argument used to initialize SYCL 2020
+    // specialization constants, `specialization_constants_buffer`, should
+    // have corresponding entry in integration header. This argument is
+    // only generated when target has no native support for specialization
+    // constants.
+    ASTContext &Context = SemaRef.getASTContext();
+    llvm::Triple T = Context.getTargetInfo().getTriple();
+    if (T.isSPIR() && T.getSubArch() == llvm::Triple::NoSubArch)
```

Comment:
3x now :)

Reply:
Haha yes! Each handler has different support for native vs native support.